File: app/src/main/java/quangnam/com/sample/base/IBasePresenter.java
package quangnam.com.sample.base;

/**
 * Created by quangnam on 11/23/17.
 * Project Sample
 */

public interface IBasePresenter<V extends IBaseView> {

    void onAttach(V view);
    void onDetach();
    boolean isViewAttached();
}

File: app/src/main/java/quangnam/com/sample/base/IBasePresenter.kt
package quangnam.com.sample.base

/**
 * Created by quangnam on 11/23/17.
 * Project Sample
 */

interface IBasePresenter<V : IBaseView> {
    val isViewAttached: Boolean

    fun onAttach(view: V)
    fun onDetach()
}
